lib/Color.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * A simple type to ensure that a hex color string takes somewhat the shape
+ * of a hex string.
+ *
+ * While it would be nice to limit the characters in the string to only valid
+ * hex characters, it seems the typescript compiler went on strike when such
```

Comment:
Found an open discussion on this https://github.com/microsoft/TypeScript/issues/41160